src/app.ts

```diff
@@ -53,24 +54,31 @@ export function generateServerlessRouter(
         try {
             const requestInformation = getRequestInformation(req.method, req.path);
             const accessToken: string = cleanAuthHeader(req.headers.authorization);
-            const isAllowed: boolean = await fhirConfig.auth.authorization.isAuthorized({
+            await fhirConfig.auth.authorization.isAuthorized({
                 ...requestInformation,
                 accessToken,
             });
-            if (isAllowed) {
-                next();
-            } else {
-                res.status(403).json({ message: 'Forbidden' });
-            }
+            res.locals.requesterUserId = fhirConfig.auth.authorization.getRequesterUserId(accessToken);
+            next();
         } catch (e) {
-            res.status(403).json({ message: `Forbidden. ${e.message}` });
+            next(e);
         }

```

Comment:
Do we have to do this try/catch? wouldn't our error mapper catch it?

Reply:
Yep, the try catch is needed. I was testing the code without the try catch with crucible. Crucible tried to do a GET on an invalid resource, example /Foo/1. Auth threw an error but the error was not caught by the error mapper. The reason is because we need to pass the error object to next. Passing the error object to next allows the next interceptor to receive the error object